pints/_mcmc/_slice_doubling.py

```diff
@@ -0,0 +1,583 @@
+                if (self._d and self._current_log_y >= self._fx_l_hat and
+                        self._current_log_y >= self._fx_r_hat):
+
+                    # Shrink the interval ``I=(l,r)``
+                    if (self._proposed[self._active_param_index] <
+                            self._current[self._active_param_index]):
+                        self._l = self._proposed[self._active_param_index]
+                        self._temp_l[self._active_param_index] = self._l
+                    else:
+                        self._r = self._proposed[self._active_param_index]
+                        self._temp_r[self._active_param_index] = self._r
+
+                    # Reset variables
+                    self._continue_check = False
+                    self._d = False
+                    self._fx_r_hat = None
+                    self._fx_l_hat = None
+
+                    return None
+
+                # If the rejection condition is not met, we continue the
+                # ``Acceptance Check`` while (r_hat - l_hat ) > 1.1*w
+                return None
+
+            # We have updated successfully the parameter of the new sample!
+
+            self._first_expansion = True
+            self._interval_found = False
+
+            # Reset active parameter index
+            if self._active_param_index == len(self._proposed) - 1:
+
+                self._active_param_index = 0
+
+                # The accepted sample becomes the new current sample
+                self._current = np.array(self._proposed, copy=True)
+
+                # The log_pdf of the accepted sample is used to construct the
+                # new slice
+                self._current_log_pdf = fx
+
+                # Sample new log_y used to define the next slice
+                self._e = np.random.exponential(1)
+                self._current_log_y = self._current_log_pdf - self._e
+
+                # Return the accepted sample
+                return np.array(self._proposed, copy=True)
+
+            else:
+                self._active_param_index += 1
+                return None
+
+        # If the trial point is rejected in the ``Threshold Check``, shrink
+        # the interval
+        if (self._proposed[self._active_param_index] <
+                self._current[self._active_param_index]):
+            self._l = self._proposed[self._active_param_index]
+            self._temp_l[self._active_param_index] = self._l
+        else:
+            self._r = self._proposed[self._active_param_index]
+            self._temp_r[self._active_param_index] = self._r
+
+        # If the point has been rejected, set flag for ``Acceptance Check``
+        # check to False so that we can sample a new point
+        self._init_check = False
+        self._continue_check = False
+
+        return None
+
+    def name(self):
+        """ See :meth:`pints.MCMCSampler.name()`. """
+        return 'Slice Sampling - Doubling'
+
+    def set_w(self, w):
```

Comment:
For parameters that have a well-defined meaning, we use a long name.

So `set_width()` or `set_learning_rate()`.

Occasionally there's no way around it and we end up with a `set_epsilon`, but this should be avoided wherever possible

Reply:
@MichaelClerx Hi Michael! Thanks for the comment. For many parameters (such as ``w``, ``m``, ``p``, ``k``,  ``j``) I used the notation presented in Neal's original paper, thinking that a user could refer to the paper and to the relevant pseudocode to have a deeper understanding of their functioning and impact on the performance of the samplers (if needed). Would you like me to change the name of some variables anyway or do you think that preserving the same notation as the original paper would be a good idea?

Reply:
The internal variables can be whatever you like, but wherever possible the method and class names should be something descriptive. You can always refer to the original notation in the docstring, e.g.

```
set_learning_rate(self, eta):
    """
    Sets the learning rate (eta in the original paper [1], see equation 17)
```